src/weights.jl

```diff
@@ -337,8 +364,12 @@ end
 
 for w in (AnalyticWeights, FrequencyWeights, ProbabilityWeights, Weights)
     @eval begin
-        Base.isequal(x::$w, y::$w) = isequal(x.sum, y.sum) && isequal(x.values, y.values)
-        Base.:(==)(x::$w, y::$w)   = (x.sum == y.sum) && (x.values == y.values)
+        Base.isequal(x::$w, y::$w) =
+            (x.values isa SubArray || y.values isa SubArray || isequal(x.values, y.values)) &&
+                isequal(x.values, y.values)
+        Base.:(==)(x::$w, y::$w)   =
+            (x.values isa SubArray || y.values isa SubArray || x.sum == y.sum) &&
```

Comment:
this is the kind of use of `x.sum` that I understand we should avoid (in this case it is OK, but I would assume that we should always use `sum(x)` etc. for safety since sometimes we do not have computed the `:sum` field correctly). Right?

Reply:
Same as above.

Reply:
I've changed more uses of `x.sum` to `sum(x)` as they were not justified.